src/lib/entry-point-transforms.ts

```diff
@@ -36,9 +37,34 @@ export const transformSources: BuildStep =
     // 1. TWO-PASS TSC TRANSFORMATION
     prepareTsConfig(args);
 
-    // First pass: collect templateUrl and styleUrls referencing source files.
-    log.info('Extracting templateUrl and styleUrls');
-    collectTemplateAndStylesheetFiles(args);
+    // First pass: extract templateUrl and styleUrls, analyse dependencies
+    analyseSourceFiles(args);
+
+    // XX: here starts the real work :-)
+    if (entryPoint.dependendingOnEntryPoints.length > 0) {
+      log.info(`${entryPoint.moduleId} depends on ${entryPoint.dependendingOnEntryPoints.join(', ')}.`);
+
+      // Were the depending entry points already built?
+      const unsatisfiedDependencies = entryPoint.dependendingOnEntryPoints
```

Comment:
I might be missing something here, but wouldn't this be easier if you construct the build graph, determine the order that things should be built, then build them in that order without checking whether or not dependencies have already been built? I think doing so would remove the need to code up build resuming of specific entry points.